### DISCUSSION TITLE
Flatten an array of objects to Map

### DISCUSSION THREAD

**Discussion by cvsudheer108** [schibsted/jslt#242] [General]
Hi JSLT Team,
I have this input:
[{
  "schema" : "http://schemas.schibsted.io/thing/pulse-simple.json#1.json",
  "id" : "1",
  "published" : "2017-05-04T09:13:29+02:00",
  "type" : "View1",
  "environmentId" : "urn:schibsted.com:environment:uuid",
  "url" : "http://www.aftenposten.no/"
},
{
  "schema" : "http://schemas.schibsted.io/thing/pulse-simple.json#1.json",
  "id" : "2",
  "published" : "2017-05-04T09:13:29+02:00",
  "type" : "View2",
  "environmentId" : "urn:schibsted.com:environment:uuid",
  "url" : "http://www.aftenposten.no/"
}
]

 And I want to get a Map back like this:

{
    "1" : {
    "schema" : "http://schemas.schibsted.io/thing/pulse-simple.json#1.json",
    "id" : "1",
    "published" : "2017-05-04T09:13:29+02:00",
    "type" : "View1",
    "environmentId" : "urn:schibsted.com:environment:uuid",
    "url" : "http://www.aftenposten.no/"
        },
  "2" : {
  "schema" : "http://schemas.schibsted.io/thing/pulse-simple.json#1.json",
  "id" : "2",
  "published" : "2017-05-04T09:13:29+02:00",
  "type" : "View2",
  "environmentId" : "urn:schibsted.com:environment:uuid",
  "url" : "http://www.aftenposten.no/"
      }
}

How do I get that done with JSLT??

Thanks!

**Comment by larsga** [comment]
Which part of this are you having difficulties with?

**Reply by cvsudheer108** [reply]
I see with JSLT, I can only transform an array of objects to another array of objects with transformation.
For eg:
let supplierTariffDetailsMap = [for(.) 
                let key = .id
                let value = .
                {$key : $value}
              ]

$supplierTariffDetailsMap

But I want to create a look-up map as shown above, so that I can look up using get-key().

**Reply by larsga** [reply]
Note that the code you wrote there is far better written as:

```
[for(.)
let key = .id
let value = .
{$key : $value}
]
```

That's also what the optimizer will turn it into.

But that's just a side issue. As to your actual question, you also have [for expressions for objects](https://github.com/schibsted/jslt/blob/master/tutorial.md#object-for-expressions). So you can write:

```
{for(.) .id : .}
```

I recommend trying this out with your input in the playground and then iterating from there.

**Reply by cvsudheer108** [reply]
Woh.. amazing.. I never knew that. I worked perfectly.. Thanks @larsga!!
